src/keystore.js

```diff
@@ -42,20 +46,40 @@ class Keystore {
 
   async close () {
     if (!this._store) return
+    await this._upgraded
     await this._store.close()
   }
 
+  // upgrade level-js to version 5.0.0
+  // https://github.com/Level/level-js/blob/master/UPGRADING.md#500
+  async _upgrade () {
+    const db = reachdown(this._store, 'level-js', true)
+    if (db && db.upgrade) {
+      if (this._upgraded) { return }
+      this._upgraded = new Promise((resolve, reject) => {
+        db.upgrade(function (err) {
+          if (err) reject(err)
+          resolve()
+        })
+      })
+      await this._upgraded
+    }
+  }
+
   async hasKey (id) {
     if (!id) {
       throw new Error('id needed to check a key')
     }
-    if (this._store.status && this._store.status !== 'open') {
+    if (!this._store || !this._store.isOpen()) {
```

Comment:
thinking about changing this by removing the `!this._store` to match the previous code behavior a bit more

Reply:
Since our dependencies use optional chaining, requiring us to use node 14+, you could use that here as `!this._store?.isOpen()`

Reply:
we have to revert this change as it messes with the [orbit-db-identity-provider](https://github.com/orbitdb/orbit-db-identity-provider/pull/72) tests

Reply:
shoot ok